.github/workflows/docker.yml

```diff
@@ -0,0 +1,61 @@
+on:
+  workflow_dispatch:
+    inputs:
+      image-prefix:
+        description: "gets suffixed with 'base' and 'sdk' to create actual image name"
+        default: ghcr.io/viamrobotics/cpp-
+      dockerfile:
+        default: Dockerfile.debian.bullseye
+      tag:
+        default: bullseye-amd64
+      build-base:
+        description: "whether to build the base image. the base images change less often and may not be necessary to rebuild."
+        type: boolean
+        default: false
+      build-sdk:
+        description: "whether to build the SDK image. if this is true and no corresponding base image exists, the job will fail."
+        type: boolean
+        default: false
+      push:
+        description: "whether to push the images after building them"
```

Comment:
just for my own understanding, In what cases would we run this workflow without pushing images to GHCR? why do we have the option to disable pushing `inputs.push` here?

Reply:
push=false for testing only; for commands with a big side effect (in this case overwriting a public image), I like to have a 'dry run' option